metomi/rose/macros/rule.py

```diff
@@ -388,8 +387,16 @@ def _process_rule(
         sci_num_count = -1
 
         # any/all processing.
+        # n.b. Because we've made "template variables" (with space) the
+        # section header at Rose 2 we need to hack the rose mini language
+        # to allow a space in this one case:
```

Comment:
Sorry but as someone not familiar with this bit of rose, I still don't understand why do space -> underscore followed by underscore -> space (and even once again space -> underscore down below)?

Reply:
So the reason for the back-and-forth is to hack around a regex not supporting spaces followed by a config lookup which has to have the exact setting name?

Reply:
Yes, however, I think it's probably deeper than just a single Regex, the entire expression is executed in Jinja2 (which does not allow variable names to contain spaces).

Reply:
This function is trying to parse rules into jinja2 so if you put in 

```
rule='template variables=FOO == 2'
```

you will get out (assuming that you have set `FOO=1` in your `rose-suite.conf` and are running `rose macro -V`.

```
rule='{% if this == 2 %}True{% else %}False{% endif %}', local_map={'this': 1.0}
```

IMO this function would be easier to understand with unit tests showing its working. I'm not going to spend a lot of time on it but I'll have a quick look.